toolkit/tools/pkg/imagecustomizerlib/customizetdnfsnapshot.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+func setSnapshotTimeInTdnfConfig(imageChroot *safechroot.Chroot, snapshotTime imagecustomizerapi.PackageSnapshotTime) error {
+	if snapshotTime == "" {
+		return nil
+	}
+
+	// Parse to time.Time and convert to epoch
+	layout := "2006:01:02"
+	date, err := time.Parse(layout, string(snapshotTime))
+	if err != nil {
+		return fmt.Errorf("failed to parse snapshot time: %w", err)
+	}
+
+	// Use end of day to include all packages published on that date
+	date = date.Add(23*time.Hour + 59*time.Minute + 59*time.Second)
```

Comment:
This is only 1 second off midnight the next day. This feels like a messy design. I'd be inclined to just leave it at midnight and let the user add the extra 24 hours if necessary.